crates/rpc/rpc-eth-types/src/fee_history.rs

```diff
@@ -376,12 +376,18 @@ where
             base_fee_per_blob_gas: header
                 .excess_blob_gas()
                 .and_then(|excess_blob_gas| Some(blob_params?.calc_blob_fee(excess_blob_gas))),
-            blob_gas_used_ratio: block.body().blob_gas_used() as f64 /
-                blob_params
-                    .as_ref()
-                    .map(|params| params.max_blob_gas_per_block())
-                    .unwrap_or(alloy_eips::eip4844::MAX_DATA_GAS_PER_BLOCK_DENCUN)
-                    as f64,
+            blob_gas_used_ratio: if block.body().blob_gas_used() == 0 {
+                // Prevent returning NaN when blob_params is defined with a max_blob_gas_per_block
+                // equal to zero.
+                0.0
+            } else {
+                block.body().blob_gas_used() as f64 /
+                    blob_params
+                        .as_ref()
+                        .map(|params| params.max_blob_gas_per_block())
+                        .unwrap_or(alloy_eips::eip4844::MAX_DATA_GAS_PER_BLOCK_DENCUN)
+                        as f64
+            },
```

Comment:
we have the same thing here once more:

https://github.com/paradigmxyz/reth/blob/e38e247b408bc6ad1e55f4d4f7af5b1fdced8948/crates/rpc/rpc-eth-api/src/helpers/fee.rs#L189-L190

what we can do is add a smol helper function for this that checks the max_blob_gas_per_block for 0.0

like fn checked_gas_used_ratio